.github/workflows/CheckMklDep.yml

```diff
@@ -0,0 +1,107 @@
+name: Check MKL_jll Dependency
+
+on:
+  workflow_call:
+    inputs:
+      auto_update:
+        description: "Automatically update MKL_jll.dep if out of date"
+        required: true
+        type: boolean
+        default: true
+    outputs:
+      is_up_to_date:
+        description: "Whether MKL_jll.dep is up to date"
+        value: ${{ jobs.check-mkl-dep.outputs.is_up_to_date }}
+  pull_request:
+
+jobs:
+  check-mkl-dep:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    outputs:
+      is_up_to_date: ${{ steps.check.outputs.is_up_to_date }}
+    steps:
+      # Add this step at the beginning to set the auto_update value
+      - name: Set auto_update
+        id: set_auto_update
+        run: |
+          if [[ "${{ github.event_name }}" == "push" ]]; then
+            echo "value=true" >> $GITHUB_OUTPUT
+          else
+            echo "value=${{ inputs.auto_update }}" >> $GITHUB_OUTPUT
+          fi
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract current MKL hash from Manifest
+        id: extract
+        run: |
+          CURRENT_HASH=$(awk '/\[\[deps\.MKL_jll\]\]/{flag=1;next}/\[\[/{if(flag){exit}}flag&&/uuid/{gsub(/^.*= "|".*$/,"",$0);printf "%s", $0}' Manifest.toml)
```

Comment:
I'm okay with this to be pragmatic, but noting in future we should aim to remove awkward use of awk.
Could we at least add a comment explaining what this magic line does?